daemon/cmd/daemon.go

```diff
@@ -852,7 +853,12 @@ func newDaemon(ctx context.Context, cleaner *daemonCleanup, params *daemonParams
 	// Start watcher for endpoint IP --> identity mappings in key-value store.
 	// this needs to be done *after* init() for the daemon in that function,
 	// we populate the IPCache with the host's IP(s).
-	d.ipcache.InitIPIdentityWatcher(d.ctx, params.StoreFactory)
+	if option.Config.KVStore != "" {
+		go func() {
```

Comment:
We could maybe run this inside a hive Job for better visibility and lifecycle management?